var/lib/delphix-platform/ansible/10-delphix-platform/roles/delphix-platform/tasks/main.yml

```diff
@@ -347,3 +347,38 @@
 - command: systemctl disable google-accounts-daemon.service
   when:
     - platform == "gcp"
+
+#
+# We want the ssh service to start as early as possible during boot up,
+# so we explicitly remove all dependencies of the service here. We've
+# configured the service not to use the "default" dependencies, via the
+# "override.conf" file in the delphix-platform package. Unfortuantely
+# though, we can't remove any dependencies specified using "After=" in
+# the original SSH service file, by using that same "override.conf"
+# file. Thus, to remove the "After=" dependencies, we have to create a
+# completely new unit definitation for the SSH service, that will
+# override the original unit definitation; Systemd allows us to use
+# files in "/etc/systemd" to replace files in "/lib/systemd".
+#
+# So, the strategy here is to copy the original unit definition from
+# "/lib/systemd" (the file is provided by the SSH package) to
+# "/etc/systemd", and then modify the copy to remove the "After=" line.
+# This way, if the original unit definitation is changed on upgrade,
+# we'll automatically use the new file from the new package; but we'll
+# still be able to modify the file to remove the "After=" dependencies.
+# Additionally, we don't modify the file in "/lib/systemd" directly, so
+# that we don't break "debsums -c", and it's easy to "revert" our
+# changes by simply removing our copy in "/etc/systemd".
+#
+- copy:
+    remote_src: yes
+    src: /lib/systemd/system/ssh.service
+    dest: /etc/systemd/system/ssh.service
+    owner: root
+    group: root
+    mode: 0644
+
+- lineinfile:
+    path: /etc/systemd/system/ssh.service
+    regexp: "^After="
```

Comment:
In the case the set of `After` dependencies change, we would probably want to be notified of that so that we can review the changes before removing them. I'd propose that we make a strict check for the current `After` dependencies and fail if those are not there.

Reply:
If we want to do that, I’d suggest we do this by creating a new package like we discussed. 

IMO, if we do like you describe, we can end up causing upgrade test failures anytime the ssh package changes; and that’s the wrong place to put that sort of notification. 

I think it’d be better to be notified of this sort of thing in the linux-pkg build/update job.

I’m open to seeing what others think, though.

Reply:
Okay, I'm fine with leaving it as is in that case. 

I'm still a bit torn whether this is better than having our own package since ssh is such a critical component and we'd want to be aware of what's going on with it. I did a quick test and it takes 4.5 minutes to build `openssh-server` (+1 minute to install build dependencies).

Reply:
If you want to get `openssh-server` integrated with linux-pkg, I'm OK with that approach too.

IMO, adding latency to the linux-pkg build is a tangential concern, and we shouldn't let that sway our decision about when to add new downstream packages. If that's a problem, we need to fix the architecture of that project, rather than avoid adding downstream packages.

Reply:
I see both approaches as having their pros and cons, but after thinking a bit more I think it is fine leaving this in `delphix-platform`.

The reason I'm mentioning the build time of the package is because as of right now there are 2 valid approaches and no clear-cut benefit of one over the other. If adding the package was a product requirement then we'd of course add it, even if it increased the build time by a lot.

Regarding linux-pkg, it was designed based on some assumptions (e.g. number of custom packages we will have and upper limit on the total build time of those packages). If we have a need to rethink the architecture of linux-pkg we will do it, but that would incur engineering costs and could add more complexity to the build.